packages/fetchai/skills/oracle_aggregation/behaviours.py

```diff
@@ -0,0 +1,157 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This package contains the behaviours for the oracle aggregation skill."""
+
+from typing import cast
+
+from aea.skills.behaviours import TickerBehaviour
+
+from packages.fetchai.protocols.oef_search.message import OefSearchMessage
+from packages.fetchai.skills.oracle_aggregation.dialogues import OefSearchDialogues
+from packages.fetchai.skills.oracle_aggregation.strategy import GenericStrategy
+
+
+DEFAULT_SEARCH_INTERVAL = 5.0
+
+
+class GenericSearchBehaviour(TickerBehaviour):
```

Comment:
Could this not be replaced with the generic search behaviour already implemented in other skills? Or at least subclass and only implement the diffs?

Reply:
Thanks, this is really useful. I had thought that our approach to packages was to make them somewhat standalone, and that is why I see quite a bit of duplication across packages, but it seems to be much better practice to inherit as much base functionality as possible. I'm planning to do a bit of reorganization with this in mind.

Reply:
In the end, I wasn't really able to inherit from another skill, the main reason being that the search here combined functionality from both the generic buyer and seller (each agent should register on the SOEF and also search for other AEAs who have registered). Also, it gets a bit messy to set the configuration for inherited skills because they use their own configuration.